extensions/optaplanner/deployment/src/main/java/io/quarkus/optaplanner/deployment/SolverManagerQuarkusConfig.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.optaplanner.deployment;
+
+import java.util.Optional;
+
+import org.optaplanner.core.config.solver.SolverManagerConfig;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+/**
+ * Subset of OptaPlanner's {@link SolverManagerConfig}.
+ */
+@ConfigGroup
+public class SolverManagerQuarkusConfig {
+
+    /**
+     * The number of solvers that run in parallel. This directly influences CPU consumption.
+     * Defaults to {@value SolverManagerConfig#PARALLEL_SOLVER_COUNT_AUTO}.
+     * Other options include a number or formula based on the available processor count.
+     */
+    @ConfigItem
+    Optional<String> parallelSolverCount;
```

Comment:
It only affects SolverManager during build time - those get injected, but we don't need to read this property at runtime.

Reply:
I tried doing this:

```
@ConfigRoot(name = "optaplanner", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
public class OptaPlannerQuarkusConfig {
```

But I am getting 

```
Caused by: java.lang.ClassNotFoundException: io.quarkus.optaplanner.deployment.OptaPlannerQuarkusConfig$$accessor
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
```

I don't need to read the property at runtime, so I am not even sure if BUILD_TIME as it stand isn't just fine?

Reply:
Discussed with Georgios. He agrees, these are build time properties (they get translated into optaplanner configuration during the build time, they aren't used at runtime). Renamed class to end with BuildTimeConfig instead of QuarkusConfig to signal that.

Reply:
Well, they might be build time for your code but I think it's typically values that you will want to change at runtime because it might depend on the hardware on which you deploy it.

I don't have a problem merging that as is for now but I think it's probably something we should think about for a later version (but it might be totally impractical, I don't know OptaPlanner at all).

Reply:
Agreed - or users wanting to override them with -D system properties?